src/compiler.ts

```diff
@@ -1475,6 +1477,7 @@ export class Compiler extends DiagnosticEmitter {
         signature.paramRefs,
         signature.resultRefs,
         typesToRefs(instance.additionalLocals),
+        instance.localsByIndex.map<string>(local => local.internalName),
```

Comment:
When using `local.name` instead of `.internalName`, I think the `.split("~")` below and the temporary array it produces can be avoided.